src/newrelic/collector/client_test.go

```diff
@@ -91,3 +96,65 @@ func TestUserAgent(t *testing.T) {
 		}
 	}
 }
+
+type roundTripperFunc func(*http.Request) (*http.Response, error)
+
+func (fn roundTripperFunc) RoundTrip(r *http.Request) (*http.Response, error) {
+	return fn(r)
+}
+
+func TestExecuteWhenMaxPayloadSizeExceeded(t *testing.T) {
+	// call is defined in collector_test.go
+	call.MaxPayloadSize = 0
```

Comment:
I feel that `call` and `actualData` should be defined in this test source file instead of pulling it in from the `var()` statement in `collector_test.go`.

Also does `call` need to be reset to a known state for each test function that access it or do changes in one test function not persist to the next?

Reply:
Is this acceptable?
```suggestion
	cmdPayload := "dummy-data"
	cmd := Cmd{
		Collectible: CollectibleFunc(func(auditVersion bool) ([]byte, error) {
			if auditVersion {
				return nil, nil
			}
			return []byte(cmdPayload), nil
		}),
		MaxPayloadSize: 100,
	}
	testedFn := fmt.Sprintf("client.Execute(payload: %v, MaxPayloadSize: %v)", cmdPayload, cmd.MaxPayloadSize)

```
----
Side note: It turns out that these fields of `Command` do not need to be set for client.Execute to work.
```
Name:      "collector-method",
Collector: "the-collector.com",
License:   "12345",
RunID:     "db97531",
```